crates/forge-runner/src/runtime.rs

```diff
@@ -0,0 +1,135 @@
+use std::any::Any;
+use std::collections::HashMap;
+
+use anyhow::Result;
+use blockifier::execution::common_hints::HintExecutionResult;
+
+use blockifier::execution::syscalls::hint_processor::SyscallHintProcessor;
+
+use cairo_felt::Felt252;
+use cairo_vm::hint_processor::hint_processor_definition::{
+    HintProcessor, HintProcessorLogic, HintReference,
+};
+use cairo_vm::serde::deserialize_program::ApTracking;
+use cairo_vm::types::exec_scope::ExecutionScopes;
+use cairo_vm::vm::errors::hint_errors::HintError;
+use cairo_vm::vm::errors::vm_errors::VirtualMachineError;
+use cairo_vm::vm::runners::cairo_runner::{ResourceTracker, RunResources};
+use cairo_vm::vm::vm_core::VirtualMachine;
+
+use cheatnet::execution::cheatable_syscall_handler::SyscallSelector;
+
+pub struct StarknetRuntime<'a> {
+    pub hint_handler: SyscallHintProcessor<'a>,
+}
+
+impl<'a> ResourceTracker for StarknetRuntime<'a> {
+    fn consumed(&self) -> bool {
+        self.hint_handler.context.vm_run_resources.consumed()
+    }
+
+    fn consume_step(&mut self) {
+        self.hint_handler.context.vm_run_resources.consume_step();
+    }
+
+    fn get_n_steps(&self) -> Option<usize> {
+        self.hint_handler.context.vm_run_resources.get_n_steps()
+    }
+
+    fn run_resources(&self) -> &RunResources {
+        self.hint_handler.context.vm_run_resources.run_resources()
+    }
+}
+
+impl<'a> HintProcessorLogic for StarknetRuntime<'a> {
+    fn execute_hint(
+        &mut self,
+        vm: &mut VirtualMachine,
+        exec_scopes: &mut ExecutionScopes,
+        hint_data: &Box<dyn Any>,
+        constants: &HashMap<String, Felt252>,
+    ) -> Result<(), HintError> {
```

Comment:
```suggestion
    ) -> HintExecutionResult {
```
is that correct?
would make for clearer code